src/overlays/actors/ovl_En_Okuta/z_en_okuta.c

```diff
@@ -21,63 +31,642 @@ const ActorInit En_Okuta_InitVars = {
+        EnOkuta_SpawnDust(&pos, &velocity, -0x14, globalCtx);
+        Audio_PlayActorSound2(&this->actor, NA_SE_EN_OCTAROCK_DEAD2);
+    }
+    if (func_800A56C8(&this->skelAnime, 15.0f)) {
+        EnOkuta_SpawnSplash(this, globalCtx);
+        Audio_PlayActorSound2(&this->actor, NA_SE_EN_OCTAROCK_LAND);
+    }
+    if (this->timer < 3) {
+        Actor_SetScale(&this->actor, ((this->timer * 0.25f) + 1.0f) * 0.01f);
+    } else if (this->timer < 6) {
+        Actor_SetScale(&this->actor, (1.5f - ((this->timer - 2) * 0.2333f)) * 0.01f);
+    } else if (this->timer < 11) {
+        Actor_SetScale(&this->actor, (((this->timer - 5) * 0.04f) + 0.8f) * 0.01f);
+    } else {
+        if (Math_ApproxF(&this->actor.scale.x, 0.0f, 0.0005f)) {
+            Audio_PlaySoundAtPosition(globalCtx, &this->actor.posRot.pos, 30, NA_SE_EN_OCTAROCK_BUBLE);
+            Item_DropCollectibleRandom(globalCtx, &this->actor, &this->actor.posRot.pos, 0x70);
+            for (i = 0; i < 20; i++) {
+                velocity.x = (Math_Rand_ZeroOne() - 0.5f) * 7.0f;
+                velocity.y = Math_Rand_ZeroOne() * 7.0f;
+                velocity.z = (Math_Rand_ZeroOne() - 0.5f) * 7.0f;
+                EffectSsDtBubble_SpawnCustomColor(globalCtx, &this->actor.posRot.pos, &velocity, &accel, &primColor,
+                                                  &envColor, Math_Rand_S16Offset(100, 50), 25, 0);
+            }
+            Actor_Kill(&this->actor);
+        }
+        this->actor.scale.y = this->actor.scale.z = this->actor.scale.x;
+    }
+}
+
+void EnOkuta_Freeze(EnOkuta* this, GlobalContext* globalCtx) {
+    Vec3f pos;
+    s16 temp_v1;
+
+    if (this->timer != 0) {
+        this->timer--;
+    }
+    if (this->timer == 0) {
+        EnOkuta_SetupDie(this);
+    }
+    if ((this->timer >= 64) && (this->timer & 1)) {
+        temp_v1 = (this->timer - 64) >> 1;
+        pos.y = (this->actor.posRot.pos.y - 32.0f) + (8.0f * (8 - temp_v1));
+        pos.x = this->actor.posRot.pos.x + ((temp_v1 & 2) ? 10.0f : -10.0f);
+        pos.z = this->actor.posRot.pos.z + ((temp_v1 & 1) ? 10.0f : -10.0f);
+        EffectSsEnIce_SpawnFlyingVec3f(globalCtx, &this->actor, &pos, 150, 150, 150, 250, 235, 245, 255,
+                                       (Math_Rand_ZeroOne() * 0.2f) + 1.9f);
+    }
+    Math_SmoothScaleMaxF(&this->actor.posRot.pos.y, this->actor.initPosRot.pos.y, 0.5f, 5.0f);
+}
+
+void EnOkuta_ProjectileFly(EnOkuta* this, GlobalContext* globalCtx) {
+    Vec3f pos;
+    Player* player = PLAYER;
+    Vec3s sp40;
+
+    this->timer--;
+    if (this->timer == 0) {
+        this->actor.gravity = -1.0f;
+    }
+    this->actor.initPosRot.rot.z += 0x1554;
+    if (this->actor.bgCheckFlags & 0x20) {
+        this->actor.gravity = -1.0f;
+        this->actor.speedXZ -= 0.1f;
+        this->actor.speedXZ = CLAMP_MIN(this->actor.speedXZ, 1.0f);
+    }
+    if (this->actor.bgCheckFlags & 8 || this->actor.bgCheckFlags & 1 || this->collider.base.atFlags & 2 ||
```

Comment:
In general, I suggest using parens to make conditions with `&&`s or `||`s easier to read, especially complex ones like these.
```suggestion
    if ((this->actor.bgCheckFlags & 8) || (this->actor.bgCheckFlags & 1) || (this->collider.base.atFlags & 2) ||
```